### DISCUSSION TITLE
Grid not applying to div

### DISCUSSION THREAD

**Discussion by jwaugh3** [tailwindlabs/tailwindcss#13184] [Help] [answered]
I am creating a basic webpage section with 2 columns and 1 row (with a subgrid in the first column). The problem is that the grid is being applied to the element, but it doesn't display as such. 

Making it `!important` doesn't seem to help. 

The only thing that has worked is if I use a CSS style defining the template columns such as `grid-template-columns: 50% 50%;` but I'd rather not have to do this every time I have a grid.

Here is my HTML:
```html
<div class="grid gap-4 py-4 grid-cols-2 grid-rows-1 w-[100%]" style="padding-top: 15vh; padding-bottom: 15vh;">
   <div class="grid grid-cols-1 grid-rows-3">
      <div class="text-center">
         <h1 data-slate-string="true" class="text-wrap">Header</h1>
      </div>
      <div class="text-center">
         <span data-slate-string="true" class="text-wrap">Description</span>
      </div>
   </div>
   <div class="relative h-[400px]">
      <img alt="alt text" loading="lazy" decoding="async" data-nimg="fill" class="" src="https://files.edgestore.dev/7rtk1mayu2usvwvj/publicFiles/_public/3b93aa16-8f00-4db8-897f-359faab196d9.svg" style="position: absolute; height: 100%; width: 100%; inset: 0px; color: transparent;">
   </div>
</div>
``` 

For reference, the Tailwind config content includes the path these files are in
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
}
```

Any ideas on why this grid isn't working?

**Comment by wongjn** [comment]
> The problem is that the grid is being applied to the element, but it doesn't display as such

Which grid and which element are you referring to?

Your code seems to work as expected in [Tailwind Play](https://play.tailwindcss.com/ezjS6EPmzv) – is this the same layout you see in your project?

**Reply by jwaugh3** [reply]
The parent grid is the one that isn't working. I didn't know Tailwind had that tool. that's helpful. This is what I see:
<img width="500px" src="https://github.com/tailwindlabs/tailwindcss/assets/37086504/7436d0c8-3eb2-486c-ba45-859dcf50478a" />

It does work in the playground, but I still seem to be having an issue with local development.
It doesn't look to be applying `grid-cols-2` and `grid-rows-1` for the parent grid. The classes applied are:
![image](https://github.com/tailwindlabs/tailwindcss/assets/37086504/a8c34e1e-7530-4d31-b2bb-95313ebade15)

**Reply by wongjn** [reply, marked as answer]
Consider checking:
- Tailwind has compiled since you added the classes.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).

Otherwise, consider providing a project that reproduces the unexpected behavior.

**Reply by jwaugh3** [reply]
> Tailwind has compiled since you added the classes.

This is what did it. TIL there is a safelist tailwind uses for compiling.
Good to know. Thanks for the help!

**Comment by Vijaykv5** [comment]
Hi @jwaugh3 

Do you need like this?
[Tailwind Play](https://play.tailwindcss.com/IacPCW1Vb2)
